west.yml

```diff
@@ -259,6 +261,10 @@ manifest:
     - name: zscilib
       path: modules/lib/zscilib
       revision: 0035be5e6a45e4ab89755b176d305d7a877fc79c
+    - name: thrift
+      remote: apache
+      path: modules/lib/thrift
+      revision: 9c0de2d1eb343910213c62325f73e3bb72361c22
```

Comment:
same comment as I made here: https://github.com/zephyrproject-rtos/zephyr/pull/53760#pullrequestreview-1247161700

Reply:
> same comment as I made here: [#53760 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/53760#pullrequestreview-1247161700)

@gmarull - this discussed several times at the TSC meeting and received no objections when it was proposed to include thrift as a module. It has been on the release plan for months as well. It's quite late to voice your concern unfortunately.

However, I'll address your other comments here.

> As I said multiple times, why do we need to keep adding modules in the repository that only a few applications will use?

Why did we add `sof` if few applications use it? Why did we add `nanopb` if few applications use it? The same could be said for virtually any module.

The value is in the eye of the beholder, it seems. In other words, to you any of these modules may seem useless. To others, it may be quite the opposite.

If between the lines you are questioning the value of supporting Thrift as an RPC framework for Zephyr, I can assure you, it has value. Specifically, saving thousands of person-hours maintaining thousands of boilerplate lines of code. It's useful to connect to perform RPC with Zephyr using any programming language in existence with every major OS using any transport in existence. In addition, it has directly improved POSIX support, C++ support, libc, Hashmaps, and so on.

Now, before it has even been merged, few applications use Thrift. Yes, your observation is true. However, I've already been approached by several people about it (including several from Nordic) who have said it seems _quite_ useful.

So few applications use Thrift until more applications use Thrift. I'm not saying it's the only option or the best option, but it is a _great_ option. It's still just an option.

> When using the example-application model, applications can add any module as needed, so I don't see the point of growing module list. 

All modules are optional. Feel free to clone whatever it is that you feel has value. For example, for Meta ASICs we currently require maximally 1 additional Zephyr module. We opt-out of using most Zephyr modules. However, at some point, we may opt-in to using additional ones (and I would encourage that).

Point being - it's easy to be a naysayer and want to maintain the status quo, but by expanding horizons just a little, Zephyr, it's ecosystem, and users benefit quite a lot.

Reply:
My main concern is that all these modules get cloned when you run `west update`, even if you never use them. I think that the minimum we should do is to disable them by default, so that it's clear that they are optional. Samples and tests can also stay within module code, see e.g. https://github.com/zephyrproject-rtos/zscilib/tree/master/samples. The fact we have so many modules enabld by default like lvgl, sof, nanopb, etc. is IMO a bad design.

Reply:
@gmarull - I wonder if there is a way to make less frequently used modules optional for end-users but still build them in CI. I recall that I might have said something along those lines when we added groups, but I think someone said all modules are technically optional to me as well.

It would be a better design if all modules were optional / disabled from the start until needed. E.g. if `west` prompted you to update a module when running tests or samples that depend on a given module (even better with `-y`). Sort of like "lazy module loading".

CI is the main motivator to get things upstreamed, I would guess.

Reply:
I've been saying this for years about disabling modules by default. Every time I get zephyr, I have a script I run that removes modules I have no need in using - the result is I can clone much faster and use far less space:
```
Threw canopennode out
Threw chre out
Threw edtt out
Threw fatfs out
Threw hal_altera out
Threw hal_atmel out
Threw hal_espressif out
Threw hal_ethos_u out
Threw hal_gigadevice out
Threw hal_infineon out
Threw hal_microchip out
Threw hal_nuvoton out
Threw hal_nxp out
Threw hal_openisa out
Threw hal_quicklogic out
Threw hal_renesas out
Threw hal_rpi_pico out
Threw hal_silabs out
Threw hal_st out
Threw hal_stm32 out
Threw hal_telink out
Threw hal_ti out
Threw hal_wurthelektronik out
Threw hal_xtensa out
Threw liblc3 out
Threw loramac-node out
Threw lvgl out
Threw lz4 out
Threw mipi-sys-t out
Threw nanopb out
Threw net-tools out
Threw nrf_hw_models out
Threw openthread out
Threw picolibc out
Threw sof out
Threw tflite-micro out
Threw TraceRecorderSource out
Threw trusted-firmware-m out
Threw trusted-firmware-a out
Threw tf-m-tests out
Threw psa-arch-tests out
Threw uoscore-uedhoc out
Threw zscilib out
```
Total size after getting zephyr: 969MB

There should also be `git gc` (garbage collection) ran on the zephyr repo and pushed back, it is actually 468MB but because of old unreferenced commits, is actually 556MB - almost 100MB of completely useless data that everyone pointlessly downloads for no reason with each clone!

But all of this is orthogonal to this PR and discussion of the module itself, which arguably does have for more use then e.g. pigweed. This sounded like a very interesting module @cfriedt will see if I can give it a check out sometime, thanks!

Reply:
I'll repeat the same thing I suggested when we adopted module groups to solve this problem (which I don't think ever worked). 

Fundamentally, this is a DWIMNWIS problem. "Update all modules". "Why did you update all modules when I only wanted these ones?"

Adding an `--update-as-needed` option to update only the modules required for a particular build target would be useful. It could even inform users exactly which modules to update on future builds for a given target via `west update <modules> <for> <this> <target>`.

With that, the ease-of-use factor is still there for end users and product makers / CI people can be explicit about only pulling in the modules they need.

Reply:
We already have module groups and there already exists a hal group. That solution has not been effective, and it's likely because of the DWIMNWIS disconnect. 

This file is just text. An additional 80 bytes of text in this file should be fine.

Reply:
The `hal` group has always been a waste of time, if I want a selective clone, I want to selectively clone hals, I don't want them all unless I'm creating tests for every single board. Case in point: nxp's hal, why would I pointlessly want to clone 2.5GB?

Reply:
I completely agree. Probably somewhat tangential to this PR though.